packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -72,6 +68,20 @@ const reducer: Reducer<State, Actions | DeleteItemActions> = (
         items: newItems,
       };
     }
+    case EditItemActionTypes.EditItemUpdated: {
+      const item = state.items.find((x) => x.id === action.id);
+      if (!item) {
+        return state;
+      }
+      const updatedItem =
+        item.type === 'query'
+          ? mapQueryToItem(action.payload as Query)
+          : mapAggregationToItem(action.payload as Aggregation);
+      return {
+        ...state,
+        items: [...state.items.filter((x) => x.id !== action.id), updatedItem],
```

Comment:
Total nit, but can be done in one go (instead two iterations)

```suggestion
        items: state.items.map((item) => {
          if (item.id === action.id) {
            return updatedItem;
          } else {
            return item
          }
        }),
```